### PR TITLE
argmax cpu: fast path for dim=0 reductions with two rows

### PR DESCRIPTION
**Pull request by sypsyp97**
## Summary
This PR adds a small CPU fast path in `argmax_kernel_impl` for a common shape pattern where reduction is over dim 0 and the reduced size is exactly 2 (e.g. `[2, ...] -> [...]`).

Instead of going through the generic `binary_kernel_reduce` path, it uses a narrowed iterator over non-reduced dims and directly compares the two candidate values per output element.

## Why
Issue: https://github.com/pytorch/pytorch/issues/140000

`argmax` eager CPU currently has notable overhead in some dim-reduction patterns. For the 2-row case, the reduction itself is trivial and dispatch/iterator overhead dominates. This path reduces that overhead while preserving semantics (tie-breaking, NaN handling) by reusing `ArgMaxOps`.

## Changes
- `aten/src/ATen/native/cpu/ReduceOpsKernel.cpp`
  - Add fast path in `argmax_kernel_impl` when:
    - single reduced dim
    - reduced dim is dim 0
    - one input
    - reduced size is 2
- `test/test_reductions.py`
  - Add `test_argmax_dim0_two_rows` covering:
    - contiguous case
    - non-contiguous sliced input
    - tie behavior

## Test
- `python3 -m py_compile test/test_reductions.py`

I can add a small benchmark snippet in this PR if maintainers want perf numbers checked in-tree.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

### CODE REVIEW

**Comment by sypsyp97** [comment]
Thanks for taking a look!\n\nThis PR intentionally keeps scope narrow: a CPU eager fast path only for `argmax(dim=0)` when reduced size is exactly 2. It reuses `ArgMaxOps::reduce` to preserve tie/NaN behavior, and adds contiguous + non-contiguous coverage in tests.\n\nIf helpful, I can also add a small benchmark snippet (for the issue shape pattern from #140000) in this PR.

File: test/test_reductions.py [hunk 1]
```diff
@@ -2522,6 +2522,21 @@ def test_argminmax_axis_with_dim_one(self, device):
...
+        )
+
+        # non-contiguous input should preserve tie-breaking semantics
+        y = x[:, ::2]
+        self.assertFalse(y.is_contiguous())
+        self.assertEqual(
+            y.argmax(dim=0),
+            torch.tensor([1, 0], device=device, dtype=torch.int64),
+        )
+
```
[the remark this answers is not in the document]
Comment (sypsyp97) [reply test/test_reductions.py:2539]: Good catch — added NaN and inf cases in 68ea68d. They compare dim=0 against transposed dim=1 to make sure the fast path matches generic reduction.
